xonsh/main.py

```diff
@@ -150,6 +150,10 @@ def premain(argv=None):
         setproctitle(' '.join(['xonsh'] + sys.argv[1:]))
     builtins.__xonsh_ctx__ = {}
     args, other = parser.parse_known_args(argv)
+    if not args.args and other and other[0].startswith('-'):
```

Comment:
This condition doesn't seem to work for me.  Try running a script and you'll see that all the script arguments are still treated as invalid.  For example, in the xonsh dir, 

```
$ xonsh release.xsh --hep
```

will return a error, but it the error from xonsh and not the error from release.xsh.  I bet this is a parse known args problem.